File: TextosJapones/src/main/java/org/jisho/textosJapones/database/dao/EstatisticaDao.java
package org.jisho.textosJapones.database.dao;

import org.jisho.textosJapones.controller.EstatisticaController.Tabela;
import org.jisho.textosJapones.model.entities.Estatistica;
import org.jisho.textosJapones.model.exceptions.ExcessaoBd;

import java.util.List;

public interface EstatisticaDao {

	void insert(Estatistica obj) throws ExcessaoBd;

	void update(Estatistica obj) throws ExcessaoBd;

	void delete(Estatistica obj) throws ExcessaoBd;

	Estatistica select(String kanji, String leitura) throws ExcessaoBd;

	List<Estatistica> select(String kanji) throws ExcessaoBd;

	List<Estatistica> selectAll() throws ExcessaoBd;
	
	List<Tabela> pesquisa(String pesquisa) throws ExcessaoBd;

}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/database/dao/EstatisticaDao.kt
package br.com.fenix.processatexto.database.dao

import br.com.fenix.processatexto.controller.EstatisticaController.Tabela
import br.com.fenix.processatexto.model.entities.comicinfo.ComicInfo
import br.com.fenix.processatexto.model.entities.processatexto.japones.Estatistica
import java.sql.SQLException
import java.util.*


interface EstatisticaDao : RepositoryDao<UUID?, Estatistica> {
    @Throws(SQLException::class)
    fun save(obj: Estatistica)

    @Throws(SQLException::class)
    fun delete(obj: Estatistica)

    @Throws(SQLException::class)
    fun select(kanji: String, leitura: String): Optional<Estatistica>

    @Throws(SQLException::class)
    fun select(kanji: String): MutableList<Estatistica>

    @Throws(SQLException::class)
    fun selectAll(): MutableList<Estatistica>

    @Throws(SQLException::class)
    fun pesquisa(pesquisa: String): List<Tabela>
}